zebra-chain/src/orchard/commitment.rs

```diff
@@ -26,14 +27,16 @@ use super::sinsemilla::*;
 /// Generates a random scalar from the scalar field 𝔽_{q_P}.
 ///
 /// <https://zips.z.cash/protocol/nu5.pdf#pallasandvesta>
-pub fn generate_trapdoor<T>(csprng: &mut T) -> pallas::Scalar
+pub fn generate_trapdoor<T>(csprng: &mut T) -> Result<pallas::Scalar, RandError>
 where
     T: RngCore + CryptoRng,
 {
     let mut bytes = [0u8; 64];
-    csprng.fill_bytes(&mut bytes);
-    // pallas::Scalar::from_bytes_wide() reduces the input modulo q_P under the hood.
-    pallas::Scalar::from_uniform_bytes(&bytes)
+    csprng
+        .try_fill_bytes(&mut bytes)
+        .map_err(|_| RandError::FillBytes)?;
```

Comment:
Daira suggests that the correct behaviour if a CSPRNG panics is to fail, to protect user funds:
https://github.com/ZcashFoundation/zebra/issues/6338#issuecomment-1514918706

This seems reasonable to me. @dconnolly can you make a decision on this?

If the correct behaviour is panicking, then we can revert the change to `try_fill_bytes`, and remove `RandError::FillBytes`. We can keep the error handling for invalid curve points.

Reply:
Replied here: https://github.com/ZcashFoundation/zebra/issues/6338#issuecomment-1515552707

I do not think we should panic in `zebra-chain`, as it is a library. The consumers of the library (probably `zebra-consensus` or another dependent) should handle the error, and probably panic at _that_ layer. 

We may want to document this in the rustdoc or on the shielded coinbase tickets

Reply:
Thanks, I'll leave it to you or @mpguerra to update those tickets or open a PR with updated docs.